bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -2397,4 +2403,25 @@ public static Image win32_new(Device device, int type, long handle) {
 	image.setHandleForZoomLevel(handle, image.getZoom());
 	return image;
 }
+
+// This class is only used for a workaround and will be removed again
+private static class StaticZoomUpdater implements AutoCloseable {
+	int currentNativeDeviceZoom;
+	int targetZoom;
+
+	private StaticZoomUpdater(int targetZoom) {
+		this.currentNativeDeviceZoom = DPIUtil.getNativeDeviceZoom();
+		this.targetZoom = targetZoom;
+		if (currentNativeDeviceZoom != targetZoom) {
+			DPIUtil.setDeviceZoom(targetZoom);
+		}
+	}
+
+	@Override
+	public void close() {
+		if (currentNativeDeviceZoom != targetZoom) {
+			DPIUtil.setDeviceZoom(currentNativeDeviceZoom);
+		}
+	}
```

Comment:
Even though these methods should only have any side-effects in case the rescale at runtime functionality is enabled, maybe we can add some additional guard for this since
- we are late in the release cycle and I would like to avoid any unnecessary risk of regressions, and
- even though adding it would not be "clean" as should actually be unnecessary, the whole class is a temporary workaround to be removed again soon.

Reply:
What kind of guard do you mean? something like `device instanceof Display display && display.isRescalingAtRuntime()`?

Reply:
Yes, I mean exactly that kind of guard.

Reply:
I updated it